src/code/z_message.c

```diff
@@ -223,338 +808,4212 @@ void Message_FindMessage(PlayState* play, u16 textId) {
+    if ((gSaveContext.options.language == LANGUAGE_JPN) && !play->msgCtx.textIsCredits) {
+        Message_DrawTextDefault(play, gfxP);
+    } else if (play->msgCtx.textIsCredits) {
+        Message_DrawTextCredits(play, gfxP);
+    } else {
+        Message_DrawTextNES(play, gfxP, 0);
+    }
+}
+
+void Message_DrawSceneTitleCard(PlayState* play, Gfx** gfxP) {
+    MessageContext* msgCtx = &play->msgCtx;
+    Gfx* gfx;
+
+    gfx = *gfxP;
+    Gfx_SetupDL39_Ptr(&gfx);
+
+    gDPSetCombineLERP(gfx++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0, PRIMITIVE,
+                      ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+    gDPSetTextureFilter(gfx++, G_TF_BILERP);
+    gDPSetAlphaDither(gfx++, G_AD_NOTPATTERN);
+    gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, msgCtx->textboxColorAlphaCurrent);
+    gDPSetEnvColor(gfx++, 140, 40, 160, 255);
+    gDPLoadTextureBlock(gfx++, gSceneTitleCardGradientTex, G_IM_FMT_I, G_IM_SIZ_8b, 64, 1, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                        G_TX_NOMIRROR | G_TX_WRAP, 6, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+    gSPTextureRectangle(gfx++, 0, XREG(77) << 2, 0x0500, (XREG(77) + XREG(76)) << 2, G_TX_RENDERTILE, 0, 0, 0x00CC,
+                        1 << 10);
+    gDPPipeSync(gfx++);
+    gDPSetCombineLERP(gfx++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0);
+    gDPSetEnvColor(gfx++, 0, 0, 0, 255);
+
+    if ((msgCtx->currentTextId < 0x1BB2) || (msgCtx->currentTextId >= 0x1BB7)) {
+        msgCtx->unk11FF8 = XREG(75);
+    }
+    msgCtx->unk11FFA = XREG(74);
+    Message_DrawTextNES(play, &gfx, 0);
+    *gfxP = gfx++;
 }
 
-void Message_BombersNotebookQueueEvent(PlayState* play, u8 event) {
-    MessageContext* msgCtx = &play->msgCtx;
+s16 sOcarinaSongFanfares[] = {
+    NA_BGM_OCARINA_SONATA,            // OCARINA_SONG_SONATA
+    NA_BGM_OCARINA_LULLABY,           // OCARINA_SONG_GORON_LULLABY
+    NA_BGM_OCARINA_NEW_WAVE,          // OCARINA_SONG_NEW_WAVE
+    NA_BGM_OCARINA_ELEGY,             // OCARINA_SONG_ELEGY
+    NA_BGM_OCARINA_OATH,              // OCARINA_SONG_OATH
+    NA_BGM_MAJORAS_LAIR,              // OCARINA_SONG_SARIAS
+    NA_BGM_OCARINA_TIME,              // OCARINA_SONG_TIME
+    NA_BGM_OCARINA_HEALING,           // OCARINA_SONG_HEALING
+    NA_BGM_OCARINA_EPONA,             // OCARINA_SONG_EPONAS
+    NA_BGM_OCARINA_SOARING,           // OCARINA_SONG_SOARING
+    NA_BGM_OCARINA_STORM,             // OCARINA_SONG_STORMS
+    NA_BGM_OCARINA_SUNS,              // OCARINA_SONG_SUNS
+    NA_BGM_INVERTED_SONG_OF_TIME,     // OCARINA_SONG_INVERTED_TIME
+    NA_BGM_SONG_OF_DOUBLE_TIME,       // OCARINA_SONG_DOUBLE_TIME
+    NA_BGM_OCARINA_LULLABY_INTRO_PTR, // OCARINA_SONG_GORON_LULLABY_INTRO
+    NA_BGM_OCARINA_LULLABY_INTRO_PTR, // OCARINA_SONG_WIND_FISH_HUMAN
+    NA_BGM_OCARINA_LULLABY_INTRO_PTR, // OCARINA_SONG_WIND_FISH_GORON
+};
+
+s8 sOcarinaSongFanfareIoData[] = {
+    0x35, 0x5C, 0x5D, 0x5E, 0x35,
+};
```

Comment:
```suggestion
s8 sOcarinaSongFanfareIoData[PLAYER_FORM_MAX] = {
    0x35, 0x5C, 0x5D, 0x5E, 
    0x35, // PLAYER_FORM_HUMAN
};
```
And fill in the rest of the comments.